pkg/exasol-rest-api/websocket_connection.go

```diff
@@ -0,0 +1,183 @@
+package exasol_rest_api
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"database/sql/driver"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"math/big"
+	"net/url"
+	"os/user"
+	"runtime"
+	"strconv"
+)
+
+type websocketConnection struct {
+	connProperties *ApplicationProperties
+	websocket      *websocket.Conn
+}
+
+func (connection *websocketConnection) close() error {
+	err := connection.send(&Command{Command: "disconnect"}, nil)
+	connection.websocket.Close()
+	connection.websocket = nil
+	return err
+}
+
+func (connection *websocketConnection) connect() error {
+	uri := fmt.Sprintf("%s:%d", connection.connProperties.ExasolHost, connection.connProperties.ExasolPort)
+	exaURL := url.URL{
+		Scheme: connection.getURIScheme(),
+		Host:   uri,
+	}
+	dialer := *websocket.DefaultDialer
+	dialer.TLSClientConfig = &tls.Config{InsecureSkipVerify: !connection.connProperties.UseTLS}
+	websocketConnection, _, err := dialer.DialContext(context.Background(), exaURL.String(), nil)
+	if err == nil {
+		connection.websocket = websocketConnection
+		connection.websocket.EnableWriteCompression(false)
+	}
+	return err
+}
+
+func (connection *websocketConnection) getURIScheme() string {
+	if connection.connProperties.Encryption {
+		return "wss"
+	} else {
+		return "ws"
+	}
+}
+
+func (connection *websocketConnection) executeQuery(query string) ([]byte, error) {
+	command := &SQLCommand{
+		Command: Command{"execute"},
+		SQLText: query,
+		Attributes: Attributes{
+			ResultSetMaxRows: 1000,
```

Comment:
Should we make the max result set size configurable?

Reply:
Maybe in future, but not for the first release